src/main/java/ip/Deadline.java

```diff
@@ -0,0 +1,28 @@
+package ip;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+class Deadline extends Task {
+
+    protected LocalDateTime by;
```

Comment:
Perhaps using `deadline` instead of `by` would be more intuitive as a standalone name? 
Correspondingly, the getter method could be phrased as `getDeadline()`

According to the code quality guidelines, 
> "Use Nouns for Things and Verbs for Actions"